src/mixins/object_interactivity.mixin.js

```diff
@@ -279,6 +279,32 @@
       return this;
     },
 
+    /**
+     * Clears the canvas.contextTop on top of the object.
+     * This function is used to clear pieces of contextTop where we render ephemeral effects
+     * Example: blinking cursror text selection.
+     * // TODO: discuss swapping restoreManually with a renderCallback, but think of async issues
+     * @param {Boolean} [restoreManually] true to don't restore the context after clear, in order to draw something else.
+     * @returns {CanvasRenderingContext2D|undefined} canvas.contextTop transformed on the object center
+     */
+    clearContextTop: function(restoreManually) {
```

Comment:
this for me solves what i didn't like of the ctx = this.prepareContextTop() and also streamline a bit the code that was divied in 3 methods

Reply:
i know that restoreManually seems a bit like dontClear but isn't exactly as that.

Reply:
Ok
It is better
The callback approach sounds good
Let's merge

Reply:
Maybe make jsdoc a bit better?

Reply:
some function hasn't be jsdoced, i know. but i really need to stop. Let me open an issue to come back to those things.